src/docker/engine.rs

```diff
@@ -92,21 +98,55 @@ impl Engine {
 
 // determine if the container engine is docker. this fixes issues with
 // any aliases (#530), and doesn't fail if an executable suffix exists.
-fn get_engine_type(ce: &Path, msg_info: &mut MessageInfo) -> Result<EngineType> {
-    let stdout = Command::new(ce)
+fn get_engine_info(
+    ce: &Path,
+    msg_info: &mut MessageInfo,
+) -> Result<(EngineType, Option<Architecture>, Option<ContainerOs>)> {
+    let stdout_help = Command::new(ce)
         .arg("--help")
         .run_and_get_stdout(msg_info)?
         .to_lowercase();
 
-    if stdout.contains("podman-remote") {
-        Ok(EngineType::PodmanRemote)
-    } else if stdout.contains("podman") {
-        Ok(EngineType::Podman)
-    } else if stdout.contains("docker") && !stdout.contains("emulate") {
-        Ok(EngineType::Docker)
+    let kind = if stdout_help.contains("podman-remote") {
+        EngineType::PodmanRemote
+    } else if stdout_help.contains("podman") {
+        EngineType::Podman
+    } else if stdout_help.contains("docker") && !stdout_help.contains("emulate") {
+        EngineType::Docker
     } else {
-        Ok(EngineType::Other)
-    }
+        EngineType::Other
+    };
+
+    let (command, query) = match kind {
+        EngineType::Docker | EngineType::PodmanRemote | EngineType::Podman => {
+            ("version", "-f \"{{ .Server.Os }},{{ .Server.Arch }}\"")
```

Comment:
Other engines like `lima` support this in a slightly different fashion so we should probably have it as a fallback:

```bash
nerdctl.lima version -f "{{ .Client.Os }},{{ .Client.Arch }}"
```

Docker and Podman support the same, so we should have it as a fallback if the above command fails.

Reply:
Specifically, this will help for clients that don't support a remote host idea (lima and a few others), and if a server isn't being used, the client is the correct value. If that doesn't work, we should just use the defaults.

So I think the correct approach is try server, try client, and then default to the normal values.

Reply:
This might only work if the host is Linux, so we might just want the arch is we fallback to the client.